daml-foundations/daml-ghc/src/DA/Daml/GHC/Compiler/Options.hs

```diff
@@ -107,9 +108,8 @@ basePackages = ["daml-prim", "daml-stdlib"]
 mkOptions :: Options -> IO Options
 mkOptions opts@Options {..} = do
     mapM_ checkDirExists $ optImportPath <> optPackageDbs
-    defaultPkgDbDir <- locateRunfiles (mainWorkspace </> "daml-foundations" </> "daml-ghc" </> "package-database")
-    let defaultPkgDb = defaultPkgDbDir </> "package-db_dir"
-    pkgDbs <- filterM Dir.doesDirectoryExist [defaultPkgDb, projectPackageDatabase]
+    mbDefaultPkgDb <- locateRunfiles (mainWorkspace </> "daml-foundations" </> "daml-ghc" </> "package-database" </> "package-db_dir")
```

Comment:
@moritzkiefer-da I introduced new `locateRunfiles` which returns `Either` instead of throwing an error, which I used here and renamed existing one which throws to `locateRunfilesOrThrowError` as it's easier to apply in all other use cases.